File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/channel/retrofit/ChannelRootPresetResponseBodyConverter.java
package org.dvbviewer.controller.data.channel.retrofit;


import org.dvbviewer.controller.entities.ChannelRoot;
import org.dvbviewer.controller.io.data.ChannelHandler;

import java.util.List;

import okhttp3.ResponseBody;
import retrofit2.Converter;

public class ChannelRootPresetResponseBodyConverter implements Converter<ResponseBody, List<ChannelRoot>> {

    private final ChannelHandler prefsHandler = new ChannelHandler();

    @Override public List<ChannelRoot> convert(ResponseBody value) {
        try {
            return prefsHandler.parse(value.byteStream());
        } catch (RuntimeException e) {
            throw e;
        } catch (Exception e) {
            throw new RuntimeException(e);
        } finally {
            value.close();
        }
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/channel/retrofit/ChannelRootPresetResponseBodyConverter.kt
package org.dvbviewer.controller.data.channel.retrofit


import okhttp3.ResponseBody
import org.dvbviewer.controller.data.api.handler.ChannelHandler
import org.dvbviewer.controller.data.entities.ChannelRoot
import retrofit2.Converter

class ChannelRootPresetResponseBodyConverter : Converter<ResponseBody, List<ChannelRoot>> {

    private val prefsHandler = ChannelHandler()

    override fun convert(value: ResponseBody): List<ChannelRoot>? {
        try {
            return prefsHandler.parse(value.byteStream())
        } catch (e: RuntimeException) {
            throw e
        } catch (e: Exception) {
            throw RuntimeException(e)
        } finally {
            value.close()
        }
    }

}